# ConflictWiki Data v0.1

We introduce the ConflictWiki dataset, a large collection of [Wikipedia articles on armed conflict](https://en.wikipedia.org/wiki/Category:Conflicts) featuring full text accompanied by pre-computed meaningful Longformer representations of the text on the document- and section-level. 
Each conflict is annotated with starting and end date, location, casualty counts, group strengths and conflict outcome. 
If provided on Wikipedia, entity articles feature auxiliary tabular information on languages, religion, ISO2 code and ideology. 


## Data Directory Structure

The data directory contains three subdirectory, `conflict`, `entity` and `mappings`. 
`conflict` contains conflict-related data and information.
`entity` contains entity-related data and information.
`mappings` contains the conflict-entity and entity-entity relationships. 
If there are multiple files in a directory, they all contain the same content.
We provide most data in multiple data formats (json, csv, pickle). 
The contents of each subdirectory is explained in the following.

```
|   data
    |   conflict
        |   data
                | embedding
                | text
        |   info
    |   entity
        |   data
                | embedding
                | text
        |   info
    |   mappings
        |   conflict_entity_id
        |   ally_enemy_pairs
        |   network
                |   aggr_edge_list
                |   node_list
```

### conflict

##### data

contains the retrieved conflict articles. 

- `embedding` features the articles' Longformer representations as a pickled dictionary. 
`conflict_id` is the conflict id; `entity_id` is the id of the entity as mentioned in the conflict article;
`embeddings` are the representations of the relevant entities in the article.
 We use the key `0` to represent the entire conflict article as a whole, irrespective of entities.

    ```
    {conflict_id: entity_id: embeddings}
    ```

- `text` is the raw full text. 

    ```
    {conflict_id: section_title: text}
    ```

##### info

contains all information on the conflicts as extracted from the [Wikipedia militarized conflict template](https://en.wikipedia.org/wiki/Template:Infobox_military_conflict).

`conflict_id`, `conflict_name`, `place`,  `date`, `date_start`, `date_end`, `n_belligerents`, `n_entities`, `strength`, `strength_num`, `casualties`, `casualties_num`, `commander`, `commander_num`, `result` tags.
date holds the date as a string, date_start and date_end are extracted from date and are datetime objects; <br>
n_belligerents, n_entities state the number of entities and belligerents involved in the conflict; <br>
strength and casualties estimate group strengths and losses per belligerent (dictionary keys are belligerent indeces); <br>
strength_num and casualties_num are numbers extracted from strength and casualties (dictionary keys are belligerent indeces); <br>
commander lists all military commanders per belligerent (dictionary keys are belligerent indeces); <br>
result is a brief textual summary of the outcome  <br>


### load external libraries

In [1]:
import pandas as pd
import pathlib

### Conflicts

##### data

contains the retrieved conflict articles. 

- `embedding` features the articles' Longformer representations as a pickled dictionary. 
`conflict_id` is the conflict id; `entity_id` is the id of the entity as mentioned in the conflict article;
`embeddings` are the representations of the relevant entities in the article.
 We use the key `0` to represent the entire conflict article as a whole, irrespective of entities.

    ```
    {conflict_id: entity_id: embeddings}
    ```

- `text` is the raw full text. 

    ```
    {conflict_id: section_title: text}
    ```

##### info

contains all information on the conflicts as extracted from the [Wikipedia militarized conflict template](https://en.wikipedia.org/wiki/Template:Infobox_military_conflict).

`conflict_id`, `conflict_name`, `place`,  `date`, `date_start`, `date_end`, `n_belligerents`, `n_entities`, `strength`, `strength_num`, `casualties`, `casualties_num`, `commander`, `commander_num`, `result` tags.<br>
date holds the date as a string, date_start and date_end are extracted from date and are datetime objects; <br>
n_belligerents, n_entities state the number of entities and belligerents involved in the conflict; <br>
strength and casualties estimate group strengths and losses per belligerent (dictionary keys are belligerent indeces); <br>
strength_num and casualties_num are numbers extracted from strength and casualties (dictionary keys are belligerent indeces); <br>
commander lists all military commanders per belligerent (dictionary keys are belligerent indeces); <br>
result is a brief textual summary of the outcome  <br>


In [6]:
conflict_info_file = pathlib.Path().absolute().parent.absolute() / "conflict" / "info" / "conflict_info.pkl"
conflict_info = pd.DataFrame.from_dict(pd.read_pickle(conflict_info_file), orient='index')
conflict_info.index.name="conflict_id"
conflict_info

conflict_name  \
conflict_id                                                      
22738                               Operation Enduring Freedom   
43063                                       Operation Anaconda   
46216                             Israeli–Palestinian conflict   
106346                              Battle of Mogadishu (1993)   
201936                                   2003 invasion of Iraq   
...                                                        ...   
64786838                                       RFDG Insurgency   
65268074                                Second Battle of Khara   
65431221                             2020 Nagorno-Karabakh war   
65548110     December 2007 Turkish incursion into northern ...   
66219979                                Battle of Mosul (2008)   

             n_belligerents  n_entities  \
conflict_id                               
22738                     2          64   
43063                     2          16   
46216                     2           7   
106346                    2           6   
201936                    2          17   
...                     ...         ...   
64786838                  2           3   
65268074                  2           5   
65431221                  2           9   
65548110                  2           2   
66219979                  2           4   

                                                         place  \
conflict_id                                                      
22738        Afghanistan, Philippines, Somalia, Georgia cou...   
43063        Shahi Kot Valley, Paktia Province, Islamic Sta...   
46216        Middle EastPrimarily in Israel, West Bank, Gaz...   
106346                                      Mogadishu, Somalia   
201936                                                    Iraq   
...                                                        ...   
64786838                                                Guinea   
65268074                   Khara, Nepal; Khara, Rukum District   
65431221     Nagorno-Karabakh and Armenian-occupied territo...   
65548110                                         northern Iraq   
66219979                                           Mosul, Iraq   

                                                        date  date_start  \
conflict_id                                                                
22738                     7 October 2001 - 28 December 2014   2001-10-07   
43063                                       March 1-18, 2002  2022-03-01   
46216        Mid-20th century - presentMain phase: 1964-1993  2021-03-21   
106346                                                        2021-03-21   
201936                                20 March - 1 May 2003   2022-03-20   
...                                                      ...         ...   
64786838                    2 September 2000 - 9 March 2001   2000-09-02   
65268074                                      7-8 April 2005  2021-03-21   
65431221                                     {{nowrap;  - }}  2021-03-21   
65548110               December 16, 2007 - December 26, 2007  2007-12-16   
66219979                           January 23 - May 24, 2008  2022-01-23   

               date_end                                             status  \
conflict_id                                                                  
22738        2014-12-28  Operation ended; Conflict ongoing  War in Afgh...   
43063        2021-03-21                                                      
46216        2021-03-21  Israeli–Palestinian peace proceslow-level figh...   
106346             None                                               None   
201936       2003-05-01                                               None   
...                 ...                                                ...   
64786838     2001-03-09                                               None   
65268074     2005-04-08                                               None   
65

### entity

##### data

contains the retrieved entity articles. 

- `embedding` features the articles' Longformer representations as a pickled dictionary. 
`entity_id` is the entity id; `section_title` is the title of the section;
`embeddings` are the representations of the sections in entity article

    ```
    {entity_id: section_title: embeddings}
    ```

- `text` is the raw full text. 

    ```
    {entity_id: section_title: text}
    ```

##### info

contains all information on the entities as extracted from the [Wikipedia article info box](https://en.wikipedia.org/wiki/Mali).

`entity_id`, `entity_name`, `num_conflicts`, `iso`, `language`, `religion`, `ideology` <br>
num_conflicts gives the number of conflicts, the entity is involved in; <br>

In [8]:
entity_info_file = pathlib.Path().absolute().parent.absolute() / "entity" / "info" / "entity_info.pkl"
entity_info = pd.DataFrame.from_dict(pd.read_pickle(entity_info_file), orient='index')
entity_info.index.name="entity_id"
entity_info

entity_name  num_conflicts iso  \
entity_id                                                
27019                   South Korea              6  KR   
1737554     Transit Center at Manas              1  []   
17207794                   Djibouti              3  DJ   
4913064                 New Zealand              6  NZ   
23235                      Pakistan             29  PK   
...                             ...            ...  ..   
14833748               Nepal Police              1  []   
66890      People's Liberation Army              1  []   
8862873                     Syrians              1  []   
214413            Armenian diaspora              1  []   
3057255              Bloomberg News              1  []   

                                                    language ideology  \
entity_id                                                               
27019      [Korean Sign Language, Pyojun-eo, Korean, Sout...       []   
1737554                                                   []       []   
17207794   [French language, Arabic language, Arabic, Fre...       []   
4913064                       [English, New Zealand English]       []   
23235      [English, Urdu, English language, Kashmiri lan...       []   
...                                                      ...      ...   
14833748                                                  []       []   
66890                                                     []       []   
8862873    [Turoyo language, Neo-Aramaic, Assyrian Neo-Ar...       []   
214413                                                    []       []   
3057255                                                   []       []   

                                                    religion  
entity_id                                                     
27019                              [No religion, Irreligion]  
1737554                                                   []  
17207794                                             [Islam]  
4913064    [No religion, Buddhism, Irreligion, Islam, Hin...  
23235             [Islam, Islam in Pakistan, State religion]  
...                                                      ...  
14833748                                                  []  
66890                                                     []  
8862873    [Assyrian Church of the East, Christianity, Su...  
214413                                                    []  
3057255                                                   []  

[703 rows x 6 columns]

### Mappings

##### conflict_entity_id

Contains the ids of all conflicts and the ids of all involved entities partitioned into belligerents.
In the examplatory conflict below, we have three belligerents, of which the first two are formed by two entities each 
and the third belligerent consists of only one entity

```
{conflict_id: [[entity_id, entity_id, entity_id, entity_id],[entity_id]]}
(1220919: [[1576797, 31717, 4887, 3434750], [40596311]])
```

In [9]:
conflict_entity_id_file = pathlib.Path().absolute().parent.absolute() / "mappings" / "conflict_entity_id" / "conflict_entity_id.pkl"
#conflict_entity_id = pd.read_pickle(conflict_entity_id_file)
conflict_entity_id= pd.DataFrame.from_dict(pd.read_pickle(conflict_entity_id_file), orient='index', columns = ["belligerent_1", "belligerent_2", "belligerent_3", "belligerent_4"])
conflict_entity_id.index.name="conflict_id"
conflict_entity_id

belligerent_1  \
conflict_id                                                      
22738        [27019, 1737554, 17207794, 4913064, 23235, 318...   
43063        [21241, 291717, 4913064, 31717, 11125639, 737,...   
46216                                                [9282173]   
106346               [1113713, 23235, 3607937, 31769, 3434750]   
201936       [899044, 22936, 31717, 679693, 7515928, 256597...   
...                                                        ...   
64786838                                              [469639]   
65268074                 [12593188, 171166, 5217236, 14833748]   
65431221                       [11125639, 73374, 746, 8862873]   
65548110                                            [11125639]   
66219979                                    [7515928, 3434750]   

                                                 belligerent_2 belligerent_3  \
conflict_id                                                                    
22738        [103100, 51673, 737, 30635, 27358, 21462079, 3...          None   
43063                                   [389363, 737, 4512092]          None   
46216         [24093, 12047, 24324, 13913, 53185055, 13093253]          None   
106346                                               [8089426]          None   
201936        [7515849, 26215470, 285632, 2185, 199866, 17633]          None   
...                                                        ...           ...   
64786838                                    [43667081, 298010]          None   
65268074                                               [66890]          None   
65431221          [214413, 3057255, 25391, 44596915, 10918072]          None   
65548110                                               [69680]          None   
66219979                                  [39955935, 40596311]          None   

            belligerent_4  
conflict_id                
22738                None  
43063                None  
46216                None  
106346               None  
201936               None  
...                   ...  
64786838             None  
65268074             None  
65431221             None  
65548110             None  
66219979             None  

[383 rows x 4 columns]

##### ally_enemy_pairs

Contains the ids of entity pairs, a conflict id and the relationship of the entities in the given conflict as displayed below

```
((entity_pair), conflict_id, entity_relationship)
((27019, 103100), 22738, "enemies")

In [10]:
ally_enemy_pairs_file = pathlib.Path().absolute().parent.absolute() / "mappings" / "ally_enemy_pairs" / "ally_enemy_pairs.pkl"
#ally_enemy_pairs = pd.read_pickle(ally_enemy_pairs_file)
ally_enemy_pairs= pd.DataFrame(pd.read_pickle(ally_enemy_pairs_file),columns = ["entity_pair", "conflict_id", "entity_relationship"])
ally_enemy_pairs.index.name="entity_pair_id"
ally_enemy_pairs

entity_pair  conflict_id entity_relationship
entity_pair_id                                                       
0                    (27019, 103100)        22738             enemies
1                     (27019, 51673)        22738             enemies
2                       (27019, 737)        22738             enemies
3                     (27019, 30635)        22738             enemies
4                     (27019, 27358)        22738             enemies
...                              ...          ...                 ...
17066              (25391, 44596915)     66219979             enemies
17067              (25391, 10918072)     66219979             enemies
17068           (44596915, 10918072)     66219979             enemies
17069             (7515928, 3434750)     66219979              allies
17070           (39955935, 40596311)     66219979              allies

[17071 rows x 3 columns]

##### network

Contains the precomputed, aggregated network, where nodes are entities and edges are all conflicts between any two respective entities aggregated.

###### aggr_edge_list

List of edges of shape (node_id 1, node_id 2, edge attributes)

```
(entity_id 1, entity_id 2, conflict attributes
27019,	103100,	{'label': 'enemies', 'label_discrete': 0, 'label_continuous': -1, 'n_conflicts': 1, 'conflict_ids': [22738], 'conflict_names': ['Operation Enduring Freedom']}
```

In [11]:
aggr_edge_list_file = pathlib.Path().absolute().parent.absolute() / "mappings" / "network" / "aggr_edge_list.pkl"
aggr_edge_list = pd.DataFrame(pd.read_pickle(aggr_edge_list_file),columns = ["entity_id_1", "entity_id_1", "conflict_attributes"])
aggr_edge_list.index.name="edge"
aggr_edge_list

entity_id_1  entity_id_1  \
edge                              
0            27019       103100   
1            27019        51673   
2              737        27019   
3            27019        30635   
4            27019        27358   
...            ...          ...   
10832       214413     44596915   
10833       214413     10918072   
10834        25391      3057255   
10835      3057255     44596915   
10836      3057255     10918072   

                                     conflict_attributes  
edge                                                      
0      {'label': 'enemies', 'label_discrete': 0, 'lab...  
1      {'label': 'enemies', 'label_discrete': 0, 'lab...  
2      {'label': 'enemies', 'label_discrete': 0, 'lab...  
3      {'label': 'enemies', 'label_discrete': 0, 'lab...  
4      {'label': 'allies', 'label_discrete': 1, 'labe...  
...                                                  ...  
10832  {'label': 'allies', 'label_discrete': 1, 'labe...  
10833  {'label': 'allies', 'label_discrete': 1, 'labe...  
10834  {'label': 'allies', 'label_discrete': 1, 'labe...  
10835  {'label': 'enemies', 'label_discrete': 0, 'lab...  
10836  {'label': 'enemies', 'label_discrete': 0, 'lab...  

[10837 rows x 3 columns]

###### node_list

List of nodes (node_id, node attributes)

```
(entity_id, entity attributes
3343, {'name': 'Belgium'}
```

In [12]:
node_list_file = pathlib.Path().absolute().parent.absolute() / "mappings" / "network" / "node_list.pkl"
node_list= pd.DataFrame(pd.read_pickle(node_list_file),columns = ["entity_id", "entity_attributes"])
node_list.index.name="node"
node_list

entity_id                                  entity_attributes
node                                                              
0           358                                {'name': 'Algeria'}
1           701                                 {'name': 'Angola'}
2           737                            {'name': 'Afghanistan'}
3           738                                {'name': 'Albania'}
4           746                             {'name': 'Azerbaijan'}
...         ...                                                ...
711    38264234  {'name': 'European Union Training Mission in M...
712    40479968  {'name': 'United Nations Force Intervention Br...
713    43112164  {'name': 'European Union Training Mission in S...
714    45690137      {'name': 'Grupos de autodefensa comunitaria'}
715    53185055             {'name': 'All-Palestine Protectorate'}

[716 rows x 2 columns]